zero/dg_cx_cu.cu

```diff
@@ -71,7 +48,63 @@ void gkyl_dg_cx_coll_cu(const struct gkyl_dg_cx *up,
   struct gkyl_array *prim_vars_ion, struct gkyl_array *prim_vars_neut,
   struct gkyl_array *upar_b_i, struct gkyl_array *coef_cx, struct gkyl_array *cflrate)
 {  
-  gkyl_cx_react_rate_cu_ker<<<up->conf_rng->nblocks, up->conf_rng->nthreads>>>(up->kernels, *up->conf_rng,
+  gkyl_cx_react_rate_cu_ker<<<up->conf_rng->nblocks, up->conf_rng->nthreads>>>(up->on_dev, *up->conf_rng,
     prim_vars_ion->on_dev, prim_vars_neut->on_dev, upar_b_i->on_dev, 
     up->vt_sq_ion_min, up->vt_sq_neut_min, coef_cx->on_dev, up->a, up->b);
 }
+
+gkyl_dg_cx*
+gkyl_dg_cx_new_cu(struct gkyl_dg_cx_inp *inp)
+{
+  gkyl_dg_cx *up = (struct gkyl_dg_cx*) gkyl_malloc(sizeof(*up));
+
+  up->cbasis = inp->cbasis;
+  up->pbasis_gk = inp->pbasis_gk;
+  up->pbasis_vl = inp->pbasis_vl;
+  up->conf_rng = inp->conf_rng;
+  up->conf_rng_ext = inp->conf_rng_ext;
+  up->phase_rng = inp->phase_rng;
+  up->grid = inp->grid;
+  up->mass_ion = inp->mass_ion;
+  up->mass_neut = inp->mass_neut;
+  up->type_ion = inp->type_ion;
+  up->vt_sq_ion_min = inp->vt_sq_ion_min;
+  up->vt_sq_neut_min = inp->vt_sq_neut_min;
+
+  int cdim = up->cbasis->ndim;
+  int poly_order = up->cbasis->poly_order;
+  int vdim_vl = up->pbasis_vl->ndim - cdim;
+  enum gkyl_basis_type b_type = up->pbasis_vl->b_type;
+
+  if (up->type_ion == GKYL_ION_H) {
+    up->a = 1.12e-18;
+    up->b = 7.15e-20;
+  }
+  else if (up->type_ion == GKYL_ION_D) {
+    up->a = 1.09e-18;
+    up->b = 7.15e-20;
+  }
+  else if (up->type_ion == GKYL_ION_HE) {
+    up->a = 6.484e-19;
+    up->b = 4.350e-20;
+  } 
+  else if (up->type_ion == GKYL_ION_NE) {
+    up->a = 7.95e-19;
+    up->b = 5.65e-20;
+  }
+
```

Comment:
can you put these parameters in a place common to both the .c and .cu files (probably the private header) please? otherwise someone may come here to adjust one of these numbers but do it for CPUs but not GPUs, or viceversa.
Also, a comment on where these numbers came from would be good.

Reply:
Fixed in last two commits